packages/poller/src/invoice/processInvoices.ts

```diff
@@ -14,75 +13,397 @@ import {
+    intents.map(intent => ({ params: intent, invoice }))
+  );
+
+  // Send all intents in one batch
+  let purchases: PurchaseAction[] = [];
+  try {
+    const intentResults = await sendIntents(
+      allIntents[0].invoice.intent_id,
+      allIntents.map(i => i.params),
+      { everclear, logger, cache, prometheus, chainService, web3Signer },
+      config,
+      requestId
+    );
+
+    // Create purchases maintaining the invoice-intent relationship
+    purchases = intentResults.map((result, index) => ({
+      target: allIntents[index].invoice,
+      purchase: {
+        intentId: result.intentId,
+        params: allIntents[index].params,
+      },
+      transactionHash: result.transactionHash,
+    }));
+
+    // Record metrics per invoice, properly handling split intents
+    for (const { invoice, intents } of batchedGroup.invoicesWithIntents) {
+      prometheus.recordSuccessfulPurchase({
+        origin: invoice.origin,
+        id: invoice.intent_id,
+        ticker: invoice.ticker_hash,
+        destination: intents[0].origin,
+        isSplit: intents.length > 1 ? 'true' : 'false',
+        splitCount: intents.length.toString(),
+      });
+      prometheus.recordInvoicePurchaseDuration(
+        Math.floor(Date.now()) - invoice.hub_invoice_enqueued_timestamp
+      );
+    }
+
+    logger.info(`Created purchases for batched ticker group`, {
+      requestId,
+      ticker: batchedGroup.ticker,
+      origin: batchedGroup.origin,
+      invoiceCount: batchedGroup.invoicesWithIntents.length,
+      totalIntents: batchedGroup.totalIntents,
+      invoiceIds: batchedGroup.invoicesWithIntents.map(i => i.invoice.intent_id),
+      allIntentResults: intentResults.map((result, index) => ({
+        intentIndex: index,
+        intentId: result.intentId,
+        transactionHash: result.transactionHash,
+        invoiceId: allIntents[index].invoice.intent_id,
+        params: allIntents[index].params,
+      })),
+      transactionHashes: intentResults.map(result => result.transactionHash),
+      duration: getTimeSeconds() - start,
+    });
+  } catch (error) {
+    // Record invalid purchase for each invoice in the batch
+    for (const { invoice } of batchedGroup.invoicesWithIntents) {
+      prometheus.recordInvalidPurchase(
+        InvalidPurchaseReasons.TransactionFailed,
+        { 
+          origin: invoice.origin, 
+          id: invoice.intent_id, 
+          ticker: invoice.ticker_hash 
+        }
+      );
+    }
+
+    logger.error('Failed to send intents for ticker group', {
+      requestId,
+      ticker: batchedGroup.ticker,
+      origin: batchedGroup.origin,
+      invoiceCount: batchedGroup.invoicesWithIntents.length,
+      invoiceIds: batchedGroup.invoicesWithIntents.map(i => i.invoice.intent_id),
+      error: jsonifyError(error),
+      duration: getTimeSeconds() - start,
+    });
+
+    throw error;
+  }
+
+  return {
+    purchases,
+    remainingBalances,
+    remainingCustodied,
```

Comment:
where are these used? custodied is difficult to track because you're making assumptions about what happens between intents being sent and other users' actions

Reply:
iiuc they are not being used..?

Reply:
The updated values are passed into subsequent invoices' `calculateSplitIntents` to make sure the balances and custodied needed for already-purchased invoices are reserved.